github_graphql.go

```diff
@@ -0,0 +1,69 @@
+package resource
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/shurcooL/githubv4"
+)
+
+func (g *GitHubClient) ListReleasesV4() ([]*github.RepositoryRelease, error) {
```

Comment:
Haven't had a chance to look into it, but I'm seeing some weird behaviour here

I'm using the following pipeline:

```yaml
resource_types:
- name: github-release-pr
  type: registry-image
  source:
    repository: concourse/github-release-resource
    tag: pr-alpine-102

resources:
- name: release
  type: github-release-pr
  source:
    owner: concourse
    repository: concourse
    access_token: ((access_token))

jobs:
- name: job
  plan:
  - get: release
```

Checks are giving the following versions (this is just a portion of them):

![Screen Shot 2020-11-12 at 10 50 21 AM](https://user-images.githubusercontent.com/26583442/98962756-f199fe80-24d4-11eb-99d3-98fc030acaba.png)

The timestamps seem to line up with valid releases to `concourse/concourse`, but for whatever reason, all the versions have `id: 0` and `tag: `v1.4.1`.

I'll try to debug in a bit, but if you have any ideas, that would be helpful

Reply:
not sure why it's like this, but I think in this case it supposes to run v3 because you haven't set github v4 url

Reply:
@aoldershaw I found the problem there and fix it with last pr,  the explanation is there https://github.com/kyoh86/exportloopref